api/apps/api/src/migrations/api/1638891599444-AddScenarioLocksTable.ts

```diff
@@ -0,0 +1,41 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddScenarioLocksTable1638891599444 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE TABLE "scenario_locks" (
+              scenario_id uuid NOT NULL,
+              user_id uuid NOT NULL,
+              grab_date timestamptz NOT NULL
+             );`,
+    );
+
+    await queryRunner.query(
+      `ALTER TABLE "scenario_locks" ADD CONSTRAINT scenario_locks_fk FOREIGN KEY (scenario_id) REFERENCES scenarios(id) ON DELETE CASCADE;`,
+    );
+
+    await queryRunner.query(
+      `ALTER TABLE "scenario_locks" ADD CONSTRAINT scenario_locks_fk_1 FOREIGN KEY (user_id) REFERENCES users(id) ON DELETE CASCADE;`,
+    );
+
+    await queryRunner.query(
+      `ALTER TABLE "scenario_locks" ADD CONSTRAINT scenario_locks_pk PRIMARY KEY (scenario_id, user_id);`,
+    );
```

Comment:
To keep it extra safe, we could add unique index on `scenarioId` (otherwise, formally, one scenario could have multiple locks)

Reply:
https://github.com/Vizzuality/marxan-cloud/pull/673/commits/4089893cc9412c2ea89e98dd68326c806486fa6d